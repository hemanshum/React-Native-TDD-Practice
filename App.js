import React, { useState } from "react";
import { Button, StyleSheet, View } from "react-native";
import { Text } from "react-native-paper";

const App = () => {
  const [isToggled, setIsToggled] = useState(false);

  return (
    <View style={styles.body} testID="body">
      <View style={styles.sectionContainer}>
        <Text style={styles.sectionTitle}>Hello World!</Text>
      </View>
      {isToggled && <Text>I am the toggled component</Text>}
      <Button
        title="Press me!"
        onPress={() => setIsToggled(!isToggled)}
        testID="pressMeButton"
        accessibilityLabel="Press me"
      />
    </View>
  );
};

const styles = StyleSheet.create({
  body: {
    backgroundColor: "#ffffff",
  },
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: "600",
    color: "#000000",
  },
});

export default App;
