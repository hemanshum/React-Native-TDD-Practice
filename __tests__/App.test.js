import React from "react";
import { cleanup, render, fireEvent } from "@testing-library/react-native";
import { toHaveStyle } from "@testing-library/jest-native";

import App from "../App";
afterEach(cleanup);

describe("App", () => {
  expect.extend({ toHaveStyle });

  it("should show hello world", () => {
    const helloWorldText = "Hello World!";
    const notFoundText = "Not found text";

    const { toJSON, getByText, queryByText } = render(<App />);

    const foundHelloWorldText = getByText(helloWorldText);
    const notFoundTextElement = queryByText(notFoundText);

    expect(foundHelloWorldText.props.children).toEqual(helloWorldText);
    expect(notFoundTextElement).toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });

  it("should not find a dynamic component", () => {
    const { queryByTestId } = render(<App />);
    const notFoundDynamicComponent = queryByTestId("dynamicComponent");

    expect(notFoundDynamicComponent).toBeNull();
  });

  it("should find the button via testId", () => {
    const testIdName = "pressMeButton";
    const { getByTestId } = render(<App />);
    const foundButton = getByTestId(testIdName);

    expect(foundButton).toBeTruthy();
  });

  it("should find the button via accessibilityLabel", () => {
    const accessibilityLabel = "Press me";
    const { getByA11yLabel } = render(<App />);
    const foundButton = getByA11yLabel(accessibilityLabel);

    expect(foundButton).toBeTruthy();
  });

  it("should find the button title", () => {
    const title = "Press me!";
    const { getByText } = render(<App />);
    const foundButtonTitle = getByText(title);

    expect(foundButtonTitle.props.children).toEqual(title);
  });

  it("should match the body style", () => {
    const { getByTestId } = render(<App />);
    const foundBodyElement = getByTestId("body");

    expect(foundBodyElement.props.style.backgroundColor).toEqual("#ffffff");
  });

  it("should match the sectionTitle style", () => {
    const { getByText } = render(<App />);
    const foundSectionTitle = getByText("Hello World!");

    expect(foundSectionTitle).toHaveStyle({
      fontSize: 24,
      fontWeight: "600",
      color: "#000000",
    });
  });

  it("should not find the toggled component", () => {
    const hiddenToggledText = "I am the toggled component";
    const { queryByText } = render(<App />);
    const notFoundHiddenComponent = queryByText(hiddenToggledText);

    expect(notFoundHiddenComponent).toBeFalsy();
  });

  it("should find the toggled component", () => {
    const testIdName = "pressMeButton";
    const toggledText = "I am the toggled component";
    const { getByTestId, getByText } = render(<App />);
    const foundButton = getByTestId(testIdName);

    fireEvent.press(foundButton);

    const foundToggledText = getByText(toggledText);

    expect(foundToggledText.props.children).toEqual(toggledText);
  });
});
