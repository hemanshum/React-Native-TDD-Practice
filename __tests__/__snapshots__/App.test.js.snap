// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should show hello world 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
  testID="body"
>
  <View
    style={
      Object {
        "marginTop": 32,
        "paddingHorizontal": 24,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#000000",
            "fontSize": 24,
            "fontWeight": "600",
          },
        ]
      }
    >
      Hello World!
    </Text>
  </View>
  <View
    accessibilityLabel="Press me"
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="pressMeButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press me!
      </Text>
    </View>
  </View>
</View>
`;
